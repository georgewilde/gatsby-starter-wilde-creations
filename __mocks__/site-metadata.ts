import { SiteMetadata } from '@context/site-metadata';

export const SiteMetadataMock: SiteMetadata = {
  companyName: 'Acme Co.',
  description: '',
  author: '',
};
