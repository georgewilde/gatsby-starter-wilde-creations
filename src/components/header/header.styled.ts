import { Link } from 'gatsby';
import styled from 'styled-components';

export const StyledHeader = styled.header`
  margin-bottom: 1.45rem;
  background: rebeccapurple;
`;

export const StyledInner = styled.div`
  padding: 0.5rem;
`;

export const StyledLogoText = styled.div`
  margin: 0;
  font-weight: 400;
  font-size: 2em;
`;

export const StyledLink = styled(Link)`
  color: white;
  text-decoration: none;
`;
