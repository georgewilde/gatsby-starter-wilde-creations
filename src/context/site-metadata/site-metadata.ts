import { createContext } from 'react';

export interface SiteMetadata {
  title: string;
}

const defaultState: SiteMetadata = {
  title: '',
};

export const SiteMetadataContext = createContext(defaultState);
