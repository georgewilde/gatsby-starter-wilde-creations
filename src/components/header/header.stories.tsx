import React from 'react';

import { Header } from './header';
import { SiteMetadataContext } from '@context/site-metadata';
import { SiteMetadataMock } from '@mocks';

export default {
  title: 'Simple Components/Header',
  component: Header,
};

export const Default = () => (
  <SiteMetadataContext.Provider value={SiteMetadataMock}>
    <Header />
  </SiteMetadataContext.Provider>
);
