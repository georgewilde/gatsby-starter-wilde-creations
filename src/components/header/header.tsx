import React, { FC } from 'react';

import {
  StyledHeader,
  StyledInner,
  StyledLink,
  StyledLogoText,
} from './header.styled';
import { SiteMetadataContext } from '@context/site-metadata';

export const Header: FC = () => (
  <SiteMetadataContext.Consumer>
    {(siteMetadata): JSX.Element => (
      <StyledHeader>
        <StyledInner className="container">
          <StyledLogoText>
            <StyledLink to="/" data-testid="siteTitle">
              {siteMetadata.title}
            </StyledLink>
          </StyledLogoText>
        </StyledInner>
      </StyledHeader>
    )}
  </SiteMetadataContext.Consumer>
);
