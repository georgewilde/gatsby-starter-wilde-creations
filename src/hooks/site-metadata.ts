import { graphql, useStaticQuery } from 'gatsby';

import { SiteMetadata } from '@context/site-metadata';

interface SiteMetadataResult {
  site: {
    siteMetadata: SiteMetadata;
  };
}

export const useSiteMetadata = (): SiteMetadata => {
  const { site } = useStaticQuery<SiteMetadataResult>(
    graphql`
      query SiteMetaData {
        site {
          siteMetadata {
            title
          }
        }
      }
    `
  );

  return site.siteMetadata;
};
