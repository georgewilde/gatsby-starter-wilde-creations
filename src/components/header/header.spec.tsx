import { render } from '@testing-library/react';
import React from 'react';
import { create } from 'react-test-renderer';

import { Header } from './header';
import { SiteMetadata, SiteMetadataContext } from '@context/site-metadata';

describe(`Header component`, () => {
  let siteMetadata: SiteMetadata;

  beforeEach(() => {
    siteMetadata = {
      companyName: 'Acme Co.',
      description: '',
      author: '',
    };
  });

  it(`should correctly render the site title`, () => {
    const { getByTestId } = render(
      <SiteMetadataContext.Provider value={siteMetadata}>
        <Header />
      </SiteMetadataContext.Provider>
    );

    expect(getByTestId('siteTitle')).toHaveTextContent('Acme Co.');
  });

  it(`should have a link to the home page on the site title`, () => {
    const { getByTestId } = render(<Header />);

    expect(getByTestId('siteTitle')).toHaveAttribute('href', '/');
  });

  it(`should render correctly`, () => {
    const tree = create(
      <SiteMetadataContext.Provider value={siteMetadata}>
        <Header />
      </SiteMetadataContext.Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
