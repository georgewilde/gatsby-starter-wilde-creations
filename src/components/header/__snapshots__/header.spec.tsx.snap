// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render correctly 1`] = `
.c0 {
  margin-bottom: 1.45rem;
  background: rebeccapurple;
}

.c1 {
  padding: 0.5rem;
}

.c2 {
  margin: 0;
}

.c3 {
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<header
  className="c0"
>
  <div
    className="c1 container"
  >
    <h1
      className="c2"
    >
      <a
        className="c3"
        data-testid="siteTitle"
        href="/"
      >
        test title
      </a>
    </h1>
  </div>
</header>
`;
